### PR TITLE
Added more debug logs

### PR DESCRIPTION
**Pull request by HarshDaryani896**
This PR adds more logs which would help us to further debug any issues.

Testing
Tested locally using a sample app and verified the logs.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,5 +34,9 @@
     "arrowParens": "always",
     "trailingComma": "es5",
     "singleQuote": true
+  },
```
Comment (ashetkar) [inline package.json:37]: What are these required for?

File: packages/pg/lib/client.js [hunk 2]
```diff
@@ -243,6 +248,7 @@ class Client extends EventEmitter {
         }
       }
     }
+    logger.warn("Given key " + key + "is invalid")
```
Comment (ashetkar) [inline packages/pg/lib/client.js]:
```suggestion
    logger.warn("Given topology-key " + key + " is invalid")
```

File: packages/pg/lib/client.js [hunk 3]
```diff
@@ -255,15 +261,18 @@ class Client extends EventEmitter {
...
       prevCount = Client.connectionMap.get(host)
     } else if (Client.failedHosts.has(host)) {
+      logger.debug("Removing " + host + " from failed host list")
       let serverInfo = Client.failedHosts.get(host)
       Client.hostServerInfo.set(host, serverInfo)
       Client.failedHosts.delete(host)
       Client.failedHostsTime.delete(host)
     }
     Client.connectionMap.set(host, prevCount + 1)
+    logger.info("Increasing connection count of " + host + "by 1")
```
Comment (ashetkar) [inline packages/pg/lib/client.js]:
```suggestion
    logger.debug("Increasing connection count of " + host + " by 1")
```

File: packages/pg/lib/client.js (hunk 2 above)
Comment (ashetkar) [reply packages/pg/lib/client.js]: I think in other drivers we fail in case of invalid values of these properties. But we can take it up separately.

File: packages/pg/lib/client.js [hunk 4]
```diff
@@ -477,14 +495,18 @@ class Client extends EventEmitter {
       } else {
         Client.placementInfoHostMap.set(placementInfo, [eachServer.host])
       }
+      logger.debug("Updated placementInfoHost Map " + [...Client.placementInfoHostMap])
       Client.hostServerInfo.set(eachServer.host, server)
       if (eachServer.public_ip === this.host) {
         Client.usePublic = true
       }
+      logger.debug("Updated hostServerInfo " + [...Client.hostServerInfo])
+      logger.debug("Use Public Ips set to " + Client.usePublic)
```
Comment (ashetkar) [inline packages/pg/lib/client.js]:
```suggestion
      logger.debug("Updated hostServerInfo to " + [...Client.hostServerInfo]) + " and usePublic to " + Client.usePublic)
```

File: packages/pg/lib/client.js [hunk 5]
```diff
@@ -1127,6 +1168,7 @@ class Client extends EventEmitter {
       if (this.connectionParameters.loadBalance) {
         let prevCount = Client.connectionMap.get(this.host)
         if (prevCount > 0) {
+          logger.debug("Decreasing connection count of " + this.host + " by 1")
```
Comment (ashetkar) [inline packages/pg/lib/client.js]:
```suggestion
          logger.debug("Decreasing connection count (" + prevCount + ") of " + this.host + " by 1")
```

File: packages/pg/lib/logger.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
+'use strict'
```
Comment (ashetkar) [inline packages/pg/lib/logger.js:1]: Why is this file needed?

File: yarn.lock [hunk 7]
```diff
@@ -2,46 +2,60 @@
 # yarn lockfile v1
 
 
-"@babel/code-frame@^7.0.0":
```
Comment (ashetkar) [inline yarn.lock:5]: Are these expected to be updated?

File: packages/pg/lib/client.js (hunk 2 above)
Comment (HarshDaryani896) [reply packages/pg/lib/client.js]: That happens for this driver also.

File: yarn.lock (hunk 7 above)
Comment (HarshDaryani896) [reply yarn.lock:5]: Yes, since we have added a new dependency `winston`, it is expected that the yarn.lock file is updated to include winston and its dependencies (adding new or updating existing).

File: packages/pg/lib/logger.js (hunk 6 above)
Comment (HarshDaryani896) [reply packages/pg/lib/logger.js:1]: It sets up the logger: setting a default log level(`info`) and allowing users to change the log level either by `LOG_LEVEL` enviornment variable or `setLogLevel()` method.

File: package.json (hunk 1 above)
Comment (HarshDaryani896) [reply package.json:37]: `winston` is added as a dependency, it is a logger library that has been used. Removing `yarn`

**Review by ashetkar — APPROVED** [review]
